### ISSUE TITLE
training and optimizer variants/techniques

### ISSUE THREAD

**Issue by roodriigoooo** [roodriigoooo/llm_playground#7]
cosine decay, and lr_warmup (already exp), gradient clipping, cyclical LR, one-cycle policy, etc. point to, present and discuss papers. 

adamW, RAdam, Lookahead, AdaBelief, etc. point to, present and discuss papers.
